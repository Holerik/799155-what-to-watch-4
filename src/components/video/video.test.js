// video.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Video from './video.jsx';

const movie = {
  id: 1,
  title: `Joker`,
  poster: `img/joker.jpg`,
  altPoster: `Joker poster`,
  background: `img/joker-bg.jpg`,
  altBackground: `Gotham City`,
  genre: [`Thriller`, `Crime`, `Drama`],
  year: 2019,
  duration: `2h 2min`,
  age: `18+`,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: {
    score: `4.3`,
    level: `very good`,
    count: 150,
  },
  director: `Todd Phillips`,
  starring: [`Joaquin Phoenix`, `Zazie Beetz`, `Robert De Niro`, `Bryan Callen`, `Shea Whigham`, `Frances Conroy`, `Glenn Fleshler`, `Brett Cullen`, `Marc Maron`],
  description: `The origin tale of the Joker (Joaquin Phoenix) – one of the most iconic villains in comic book history.`,
  review: `Arthur Fleck (Joaquin Phoenix) isn’t happy with his life. He struggles to make money as a part-time clown while sharing a rundown apartment with his ailing mom (Frances Conroy). But Arthur lives in a city struck by hard times where a decent, honest living is difficult to come by. He also suffers from a condition that causes him to break into uncontrollable laughter. None of this stops Arthur from dreaming big. He aspires to be a stand-up comedian and attempts to write jokes in his diary. Caught in between it all, Arthur slowly begins to lose his grip on sanity`,
};

const MockComponent = () => {
  return (
    <div>
      <Video
        isPlaying={false}
        src={movie.preview}
        isMuted={true}
        poster={movie.poster}
        width={280}
        onExitButtonClick={() => {}}
      />
    </div>
  );
};

describe(`Video tests`, () => {
  it(`Moviecard should rendered as video`, () => {
    const tree = renderer
    .create((
      <MockComponent
      />), {
      createNodeMock() {
        return {};
      }
    }
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

