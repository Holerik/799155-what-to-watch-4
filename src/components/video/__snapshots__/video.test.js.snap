// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video tests Moviecard should rendered as video 1`] = `
<div>
  <div
    className="player"
  >
    <button
      className="player__exit visually-hidden"
      onClick={[Function]}
      type="button"
    >
      Exit
    </button>
    <div
      className="player__controls visually-hidden"
    >
      <div
        className="player__controls-row"
      >
        <div
          className="player__time"
        >
          <progress
            className="player__progress"
            max="100"
            value="0"
          />
          <div
            className="player__toggler"
            style={
              Object {
                "left": "0%",
              }
            }
          >
            Toggler
          </div>
        </div>
        <div
          className="player__time-value"
        >
          
        </div>
      </div>
      <div
        className="player__controls-row"
      >
        <button
          className="player__play"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <use
              xlinkHref="#play-s"
            />
          </svg>
          <span>
            Pause
          </span>
        </button>
        <div
          className="player__name"
        >
          Transpotting
        </div>
        <button
          className="player__full-screen"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlinkHref="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
    <video
      className="player__video"
      muted={true}
      poster="img/joker.jpg"
      width={280}
    >
      <source
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      />
    </video>
  </div>
</div>
`;
